test/common.py

```diff
@@ -388,6 +388,72 @@ def metafunc(self, wasmfs, *args, **kwargs):
   return metafunc
 
 
+def also_with_nodefs(func):
+  @wraps(func)
+  def metafunc(self, fs, *args, **kwargs):
+    if DEBUG:
+      print('parameterize:fs=%s' % (fs))
+    if fs == 'nodefs':
+      self.setup_nodefs_test()
+    else:
+      self.emcc_args += ['-DMEMFS']
+      assert fs is None
+    func(self, *args, **kwargs)
+
+  parameterize(metafunc, {'': (None,),
+                          'nodefs': ('nodefs',)})
+  return metafunc
+
+
+def also_with_nodefs_both(func):
+  @wraps(func)
+  def metafunc(self, fs, *args, **kwargs):
+    if DEBUG:
+      print('parameterize:fs=%s' % (fs))
+    if fs == 'nodefs':
+      self.setup_nodefs_test()
+    elif fs == 'rawfs':
+      self.require_node()
+      self.emcc_args += ['-DNODERAWFS']
+      self.set_setting('NODERAWFS')
+    else:
+      self.emcc_args += ['-DMEMFS']
+      assert fs is None
+    func(self, *args, **kwargs)
+
+  parameterize(metafunc, {'': (None,),
+                          'nodefs': ('nodefs',),
+                          'rawfs': ('rawfs',)})
+  return metafunc
+
+
+def with_all_fs(func):
+  @wraps(func)
+  def metafunc(self, fs, *args, **kwargs):
+    if DEBUG:
+      print('parameterize:fs=%s' % (fs))
+    if fs == 'nodefs':
+      self.setup_nodefs_test()
+    elif fs == 'rawfs':
+      self.require_node()
+      self.emcc_args += ['-DNODERAWFS']
+      self.set_setting('NODERAWFS')
+    elif fs == 'wasmfs':
+      self.set_setting('WASMFS')
+      self.set_setting('FORCE_FILESYSTEM')
+      self.emcc_args += ['-DWASMFS']
+    else:
+      self.emcc_args += ['-DMEMFS']
+      assert fs is None
+    func(self, *args, **kwargs)
+
+  parameterize(metafunc, {'': (None,),
+                          'nodefs': ('nodefs',),
+                          'rawfs': ('rawfs',),
+                          'wasmfs': ('wasmfs',)})
```

Comment:
This won't test wasmfs with nodefs/rawfs, will it? Those should be orthogonal I think. That is, a test could do
```python
@also_with_wasmfs
@also_with_nodefs_both
```

Reply:
Makes sense. How about adding a TODO to remove this and use the two separate decorators, to get full coverage?

Reply:
(in particular, we should avoid adding more usages of this decorator)

Reply:
Added a comment.